### PR TITLE
Service layer mising methods

### PR DESCRIPTION
**Pull request by chaenni**
- missing service layer methods
- insert reservation now inserts the reservation into the DB

### CODE REVIEW

File: AutoReservation.BusinessLayer/CarManager.cs [hunk 1]
```diff
@@ -36,12 +36,23 @@ public T GetCarById<T>(int id)
                 }
                 else
                 {
-                    throw new CarTypeMismatchException($"Car should be {nameof(T)} but is actually {nameof(car)}");
+                    throw new CarTypeMismatchException();
```
Comment (thde) [inline AutoReservation.BusinessLayer/CarManager.cs]: Why this?

File: AutoReservation.Service.Wcf/CarReservationService.cs [hunk 2]
```diff
@@ -23,5 +38,59 @@ public CustomerDto DeleteCustomer(CustomerDto customer)
...
+
+        public CarDto InsertCar(CarDto customer)
+        {
+            var entity = DtoConverter.ConvertToEntity(customer);
+            var result = carManager.InsertCar(entity);
+            return DtoConverter.ConvertToDto(result);
+        }
+
+        public CarDto GetCarById(int Id)
+            => DtoConverter.ConvertToDto(carManager.GetCarById<Car>(Id));
```
Comment (thde) [inline AutoReservation.Service.Wcf/CarReservationService.cs]: 🎉

File: AutoReservation.BusinessLayer/CarManager.cs (hunk 1 above)
Comment (lroellin) [reply AutoReservation.BusinessLayer/CarManager.cs]: Chunt vo mir. Ha ned usegfunde wieni die exakte Klassename usebechume und denn bringts wie nüt. Isch au I mim PR drin

Comment (thde) [reply AutoReservation.BusinessLayer/CarManager.cs]: I see 👍

**Pull request merged**